scio-smb/src/main/scala/com/spotify/scio/smb/syntax/SortMergeBucketSCollectionSyntax.scala

```diff
@@ -49,16 +53,20 @@ final class SortedBucketSCollection[T](private val self: SCollection[T]) {
    *   contains information about key function, bucket and shard size, etc.
    */
   @experimental
-  def saveAsSortedBucket(write: SortedBucketIO.Write[_, _, T]): ClosedTap[Nothing] = {
+  def saveAsSortedBucket(
+    write: SortedBucketIO.Write[_, _, T]
+  ): ClosedTap[T] = {
+    val beamValueCoder = self.internal.getCoder
+    implicit val valueCoder: Coder[T] = Coder.beam(beamValueCoder)
```

Comment:
There should normally be a scio coder available directly in the SCollection
```suggestion
    import self.coder
```

Reply:
awesome!